src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sortedArrayOfSides = [a, b, c].sort((el1, el2) => el2 - el1);
 
-export class Rectangle implements Figure {}
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side length can`t be 0 or less');
```

Comment:
[CODE STYLE] - When throwing an error, provide a detailed description in the error message. The current message 'Side length can't be 0 or less' is good, but consider making it more specific, such as 'Triangle side length must be greater than 0'.